src/nvim/getchar.c

```diff
@@ -1847,7 +1849,7 @@ static mapblock_T *find_typed_map(const bool timedout, const int local_State,
     return NULL;
   }
 
-  int keylen = *keylenp;
+  int keylen = 0;
   mapblock_T *mp = NULL;
   mapblock_T *mp2 = NULL;
   int nolmaplen;
```

Comment:
Wow, scratch this. I'm stupid. My "improved" version does something different, and the initial `keylen = mp_match_len` wasn't useless. Still, I think the following would be an improvement:

```c
// If no partly match found, use the longest full match.
if (keylen != KEYLEN_PART_MAP) {
  *keylenp = mp_match_len;
  return mp_match;
}
*keylenp = keylen;
return mp;
```